rp2040-hal/src/gpio/mod.rs

```diff
@@ -967,6 +967,19 @@ pub trait DefaultTypeState: crate::typelevel::Sealed {
     type PullType: PullType;
 }
 
+// Clear input enable for pins 26-29 of bank0, as these pins are ADC pins.
+// If the pins are connected to an analoug input, the signal level may not
+// be valid for a digital input.
+macro_rules! reset_ie {
+    ( Bank0, $pads:ident ) => {
+        (*$pads).gpio[26].modify(|_, w| w.ie().clear_bit());
```

Comment:
The calling macro (gpio!) uses `pads.foo()` ... why is `(*$pads)` needed here?

Reply:
Indeed. I'm not sure why I wrote it that way. I remember that I first had some experimental code without using macros, and only integrated it into the existing macros later. Probably the intermediate code was slightly different and needed it, and I just did not think about it when copying the code into the macro.